### PR TITLE
V1

### PR DESCRIPTION
**Pull request by sauvainr**
Major change: location configuration is now done by creating devices.
GPS and other location are acceptable. See the README.

### CODE REVIEW

**Review by guitarmind — COMMENTED** [review]
@sauvainr Checking, will try to test it before I go...

**Comment by sauvainr** [comment]
I made a couple of fixes this morning.. looks more stable now.

File: openweatherapi.yaml [hunk 1]
```diff
@@ -4,35 +4,63 @@ info:
...
   - name: appid
     title: API Key
     description: The API Key, provided from the OpenWeatherMap account
     type: string
+    required: true
+  - name: lang
+    title: Language
+    description: Default language
+    type: string
+    pattern: "[a-zA-Z]+(_[a-zA-Z]+)?"
```
Comment (guitarmind) [inline openweatherapi.yaml:27]: Suggest to add `en` as the default?

**Review by guitarmind — CHANGES_REQUESTED** [review]
Got the following solution error log while creating a GPS location device and click APPLY for service ocnfig:
```
2019-12-17 13:51:55   CONFIG   NOTE   [OPENWEATHERAPI]
Serviceconfig openweatherapi updated.
data: {
    "id": "28962750-2090-11ea-b2ca-039493305c37",
    "name": "o12a53xkhlssg0000_openweatherapi",
    "script_key": "openweatherapi"
  }
2019-12-17 13:51:56   SCRIPT   ERROR   [CONFIG SERVICE]
{"cod":"400","message":"Nothing to geocode"}
data: {
    "elapsed": 498,
    "location": "module/c2c.murano2cloud:21",
    "module": "c2c.murano2cloud"
  }
2019-12-17 13:51:56   SCRIPT   ERROR   [CONFIG SERVICE]
{"cod":"404","message":"city not found"}
data: {
    "elapsed": 833,
    "location": "module/c2c.murano2cloud:21",
    "module": "c2c.murano2cloud"
  }
2019-12-17 13:51:56   CALL   WARN   [OPENWEATHERAPI GETWEATHER]
service call failed
response: "{\"cod\":\"400\",\"message\":\"Nothing to geocode\"}"
data: {
    "code": 400
  }
{
    "count": 1
  }
```

*GPS Location:*
```
lon: 121.1837289
lat: 25.0344194
```

*Device Name:*
`121.1837289,25.0344194`

*Testing Product:*
https://www.exosite.io/business/wk1bfxw3f3/connectivity/o12a53xkhlssg0000/devices

File: modules/c2c/murano2cloud.lua [hunk 2]
```diff
@@ -4,25 +4,48 @@ local murano2cloud = {}
 
 murano2cloud.alias = "openweatherapi"
 
-function murano2cloud.addIdentity(identity)
+local function getQuery(identity)
+  local lon, lat = string.match(identity, '^([0-9.]+),([0-9.]+)$')
```
Comment (guitarmind) [inline modules/c2c/murano2cloud.lua]: @sauvainr This regex is not happy with the real GPS example I put, but the following works from the regex tool
```
^([0-9.]*),([0-9.]*)
```

My Example:
```
121.1837289,25.0344194
```

Comment (sauvainr) [reply modules/c2c/murano2cloud.lua]: The match works.. maybe you had an extra space? I loosed a bit the regex to parse if have a some front or back space.

**Comment by sauvainr** [comment]
I tested the GPS after the fix and seems all good now. I ll go on an merge it.
Thanks @guitarmind enjoy your honeymoon!!

**Pull request merged**